pkg/gitreceive/build.go

```diff
@@ -112,6 +110,20 @@ func build(
 		}
 	}
 
+	_, disableCaching := appConf.Values["DEIS_DISABLE_CACHE"]
+	slugBuilderInfo := NewSlugBuilderInfo(appName, gitSha.Short(), disableCaching)
+
+	if slugBuilderInfo.DisableCaching() {
+		log.Debug("caching disabled for app %s", appName)
+		// If cache file exists, delete it
+		if _, err := storageDriver.Stat(context.Background(), slugBuilderInfo.CacheKey()); err == nil {
```

Comment:
They do cost money, but it's `$0.01 / 10K` per read API request on s3 at least, so I'd consider it negligible.

So it's much cheaper to check before attempting to delete, which costs `$0.01 / 1K`